sdk/iot/hub/src/az_iot_hub_client_methods.c

```diff
@@ -135,7 +134,7 @@ AZ_NODISCARD az_result az_iot_hub_client_methods_response_get_publish_topic(
   AZ_RETURN_IF_FAILED(az_span_u32toa(remainder, (uint32_t)status, &remainder));
 
   remainder = az_span_copy(remainder, methods_response_topic_properties);
-  remainder = az_span_copy(remainder, request_id);
+  AZ_RETURN_IF_FAILED(az_span_u32toa(remainder, request_id, &remainder));
```

Comment:
This should never fail given the above `required_length` check.

Reply:
Yes but unfortunately we have no way around the `unused_return_value` compiler flag right now.

Reply:
is this not set to throw a compile warning if the ret is not checked?

Reply:
I believe this should work as a workaround.
```C
az_result unused = az_span_u32toa(remainder, request_id, &remainder);
(void)unused;
```

Reply:
This API is currently a `NO_DISCARD`. I can try ashon's suggestion and see if it passes the gate

Reply:
Unfortunately this won't work. Rid is hex-encoded, not base10.
```
20200506 170509.123 3 paho-sample-device1 <- PUBLISH msgid: 0 qos: 0 retained: 0 payload: null
Topic: $iothub/methods/POST/asd/?$rid=a
Direct Method arrived
Status: 404	Payload:{}
```